util/design/update-manifest-json.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Update the creator_manuf_state in a ROM_EXT manifest."""
+
+import argparse
+import json
+
+import hjson
+from lib.ImmutableSectionProcessor import ImmutableSectionProcessor
+
+_USAGE_CONSTRAINTS_NAME = "usage_constraints"
+_MANUF_STATE_CREATOR_NAME = "manuf_state_creator"
+_SELECTOR_BITS_NAME = "selector_bits"
+_IDENTIFIER_NAME = "identifier"
+
+_SEL_MANUF_STATE_CREATOR = (1 << 8)
+
+# This must match the definitions in chip.h.
+_CHIP_ROM_EXT_IDENTIFIER = 0x4552544f
+
+
+class RomExtImmutableSection(ImmutableSectionProcessor):
+
+    def __init__(self, rom_ext_elf, json_data):
+        super().__init__(rom_ext_elf, json_data)
+
+    def update_manifest_with_creator_manuf_state_data(self) -> None:
+        """Update the manifest with the manuf_state_creator data.
+        Args:
+            None
+        Returns:
+            None
+        """
+
+        selector_bits = self.json_data[_USAGE_CONSTRAINTS_NAME][
+            _SELECTOR_BITS_NAME
+        ]
+
+        # Ensure the selector bit of `manuf_state_creator` field is set.
+        new_selector_bits = selector_bits | _SEL_MANUF_STATE_CREATOR
```

Comment:
Shouldn't the manifest bazel rule already do this: https://cs.opensource.google/opentitan/opentitan/+/master:rules/manifest.bzl;l=81?q=manifest.bzl&ss=opentitan%2Fopentitan

Reply:
yes, the selector bit of `manuf_state_creator` field will be set by the manifest bazel rule if the `manuf_state_creator` attribute is provided.

The operation here is to handle the case that the `manuf_state_creator` attribute wasn't specified when generating the manifest json file. Alternatively,  we can assume that `manuf_state_creator` should be specified when generating the manifest json file and validate its presence by checking the selector bit here. Let me know which approach works best for you.

Reply:
Ah got it ok that makes sense to me. We can keep as is then.